crates/isograph_schema/src/validate_schema.rs

```diff
@@ -453,12 +453,13 @@ fn validate_and_transform_client_fields(
         },
     ))?;
 
-    get_all_errors_or_all_ok_iter(client_fields.into_iter().map(|client_field| {
+    get_all_errors_or_all_ok_iter(client_fields.iter().map(|client_field| {
```

Comment:
- Sorry about shipping a bunch of stuff that cleaned up `validate_schema` 😭  while you were working on it
- It looks like the reason we are passing references here is so that we can also pass `&ClientFields` down, in order to look up the nested client field's `variant`. I'm not really following. Why?
- I would imagine that what we need is to go from a client field + linked field selection to "is this a pointer or not", right? In which case, in `validate_field_type_exists_and_is_linked`, the `FieldType::ClientField(_) =>` match arm should have an enum which is `ClientFieldId` or `ClientPointerId`.
    - Let's make this enum generic and call it `FieldOrPointer<TField, TPointer>`, since we can use this enum to distinguish between scalar and linked server fields eventually.

Am I understanding correctly?